### PR TITLE
Normalize tx/harriscounty_gov

### LINKED ISSUE #652: normalize tx/harriscounty_gov
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #649

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch tx/harriscounty_gov
    poetry run vaccine-feed-ingest parse tx/harriscounty_gov
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize tx/harriscounty_gov
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by MoralCode**
# Normalize tx/harriscounty_gov

| Key Details |
|-|
| Resolves #652  |
State: tx |
Site: harriscounty_gov |

## Notes
Seems pretty standard. not much to report.
the vaccine inventory normalizer might break if it sees any value besides `Moderna` but it seems like all the sites so far only carry Moderna so that should be fine for now

## Data sample

```json
{"id": "tx_harriscounty_gov:c9709ff8-c8b9-4363-88b2-47f2910b6742", "name": "Bethel's Family Church", "address": {"street1": "12660 Sandpiper Dr.", "street2": null, "city": "Houston", "state": "TX", "zip": "77035"}, "location": {"latitude": 29.63934911, "longitude": -95.50383887}, "contact": [{"contact_type": null, "phone": "(832) 927-8787", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "https://vacstrac.hctx.net/landing", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-06-02", "closes": null}], "opening_hours": [{"day": "monday", "opens": "08:00", "closes": "12:00"}, {"day": "tuesday", "opens": "08:00", "closes": "12:00"}, {"day": "wednesday", "opens": "08:00", "closes": "12:00"}, {"day": "thursday", "opens": "08:00", "closes": "12:00"}, {"day": "friday", "opens": "08:00", "closes": "12:00"}, {"day": "saturday", "opens": "08:00", "closes": "12:00"}, {"day": "sunday", "opens": "08:00", "closes": "12:00"}], "availability": null, "inventory": null, "access": null, "parent_organization": {"id": "harris_county_public_health", "name": "Harris County Public Health"}, "links": null, "notes": null, "active": null, "source": {"source": "tx_harriscounty_gov", "id": "c9709ff8-c8b9-4363-88b2-47f2910b6742", "fetched_from_uri": "https://publichealth.harriscountytx.gov/Resources/2019-Novel-Coronavirus/Register-for-COVID-19-Vaccine", "fetched_at": "2021-06-25T16:34:33.161585", "published_at": "2021-06-15T07:13:12.069000", "data": {"attributes": {"CreationDate": 1622321276387, "Creator": "COVIDPublicHCPH", "EditDate": 1623766392069, "Editor": "COVIDPublicHCPH", "address": "12660 Sandpiper Dr., Houston, TX 77035", "agency": "Harris County Public Health", "closedatefilter": 1622934900000, "dose": "First,Second", "entrydate": "2021-06-15T09:13:09.379-05:00", "globalid": "c9709ff8-c8b9-4363-88b2-47f2910b6742", "lat": 29.63934911, "layer_id": 0, "location": "Bethel's Family Church ", "lon": -95.50383887, "notes": null, "objectid": 16, "opendate": 1622649600000, "opendate_str": "JUNE 2", "opendatefilter": 1622286900000, "operatinghours": "8:00 AM - 12:00 PM", "phone": "832-927-8787", "service_item_id": "15f20c50666a45f5be5cce3a4a655a20", "typeofvaccine": "Moderna", "vdu_no": "VDU 5", "website": "https://vacstrac.hctx.net/landing"}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -95.50383200659596, "y": 29.639344001020923}}}}
{"id": "tx_harriscounty_gov:031e6ca7-742c-4a95-8606-362217a9c00b", "name": "Gloria Dei Lutheran", "address": {"street1": "18220 Upper Bay Rd.", "street2": null, "city": "Houston", "state": "TX", "zip": "77058"}, "location": {"latitude": 29.54661299453424, "longitude": -95.09065698603666}, "contact": [{"contact_type": null, "phone": "(832) 927-8787", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "https://vacstrac.hctx.net/landing", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-06-01", "closes": null}], "opening_hours": [{"day": "monday", "opens": "08:00", "closes": "16:00"}, {"day": "tuesday", "opens": "08:00", "closes": "16:00"}, {"day": "wednesday", "opens": "08:00", "closes": "16:00"}, {"day": "thursday", "opens": "08:00", "closes": "16:00"}, {"day": "friday", "opens": "08:00", "closes": "16:00"}, {"day": "saturday", "opens": "08:00", "closes": "16:00"}, {"day": "sunday", "opens": "08:00", "closes": "16:00"}], "availability": null, "inventory": null, "access": null, "parent_organization": {"id": "harris_county_public_health", "name": "Harris County Public Health"}, "links": null, "notes": null, "active": null, "source": {"source": "tx_harriscounty_gov", "id": "031e6ca7-742c-4a95-8606-362217a9c00b", "fetched_from_uri": "https://publichealth.harriscountytx.gov/Resources/2019-Novel-Coronavirus/Register-for-COVID-19-Vaccine", "fetched_at": "2021-06-25T16:34:33.161585", "published_at": "2021-06-15T07:11:06.344000", "data": {"attributes": {"CreationDate": 1622317889463, "Creator": "COVIDPublicHCPH", "EditDate": 1623766266344, "Editor": "COVIDPublicHCPH", "address": "18220 Upper Bay Rd., Houston TX 77058", "agency": "Harris County Public Health", "closedatefilter": 1622937600000, "dose": "First,Second", "entrydate": "2021-06-15T09:11:02.085-05:00", "globalid": "031e6ca7-742c-4a95-8606-362217a9c00b", "lat": null, "layer_id": 0, "location": "Gloria Dei Lutheran", "lon": null, "notes": null, "objectid": 34, "opendate": 1622563200000, "opendate_str": "JUNE 1", "opendatefilter": 1622275200000, "operatinghours": "8:00 AM - 4:00 PM", "phone": "832-927-8787", "service_item_id": "15f20c50666a45f5be5cce3a4a655a20", "typeofvaccine": "Moderna", "vdu_no": "VDU 3", "website": "https://vacstrac.hctx.net/landing"}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -95.09065698603666, "y": 29.54661299453424}}}}
{"id": "tx_harriscounty_gov:4d5f3e9b-e6e0-4d9c-8985-c1245343a518", "name": "Clay Road Community Center", "address": {"street1": "18650 Clay Rd.", "street2": null, "city": "Houston", "state": "TX", "zip": "77084"}, "location": {"latitude": 29.83111367, "longitude": -95.69012199}, "contact": [{"contact_type": null, "phone": "(832) 927-8787", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "https://vacstrac.hctx.net/landing", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-06-01", "closes": null}], "opening_hours": [{"day": "monday", "opens": "13:00", "closes": "19:00"}, {"day": "tuesday", "opens": "13:00", "closes": "19:00"}, {"day": "wednesday", "opens": "13:00", "closes": "19:00"}, {"day": "thursday", "opens": "13:00", "closes": "19:00"}, {"day": "friday", "opens": "13:00", "closes": "19:00"}, {"day": "saturday", "opens": "13:00", "closes": "19:00"}, {"day": "sunday", "opens": "13:00", "closes": "19:00"}], "availability": null, "inventory": null, "access": null, "parent_organization": {"id": "harris_county_public_health", "name": "Harris County Public Health"}, "links": null, "notes": null, "active": null, "source": {"source": "tx_harriscounty_gov", "id": "4d5f3e9b-e6e0-4d9c-8985-c1245343a518", "fetched_from_uri": "https://publichealth.harriscountytx.gov/Resources/2019-Novel-Coronavirus/Register-for-COVID-19-Vaccine", "fetched_at": "2021-06-25T16:34:33.161585", "published_at": "2021-06-15T07:12:15.075000", "data": {"attributes": {"CreationDate": 1622321276387, "Creator": "COVIDPublicHCPH", "EditDate": 1623766335075, "Editor": "COVIDPublicHCPH", "address": "18650 Clay Rd., Houston, TX 77084", "agency": "Harris County Public Health", "closedatefilter": 1622932200000, "dose": "First,Second", "entrydate": "2021-06-15T09:12:12.353-05:00", "globalid": "4d5f3e9b-e6e0-4d9c-8985-c1245343a518", "lat": 29.83111367, "layer_id": 0, "location": "Clay Road Community Center", "lon": -95.69012199, "notes": null, "objectid": 44, "opendate": 1622563200000, "opendate_str": "JUNE 1", "opendatefilter": 1622280600000, "operatinghours": "1:00 PM - 7:00 PM", "phone": "832-927-8787", "service_item_id": "15f20c50666a45f5be5cce3a4a655a20", "typeofvaccine": "Moderna", "vdu_no": "VDU 5", "website": "https://vacstrac.hctx.net/landing"}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -95.68846226543629, "y": 29.831190858348773}}}}
```

## Before Opening a PR
- [X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/runners/tx/harriscounty_gov/normalize.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    return addr
+
+
+def _get_opening_dates(site: dict) -> Optional[List[schema.OpenDate]]:
+
+    start_date = site["attributes"].get("opendate")
+
+    # this field does not seem to exist, but might later
+    # see schema at https://services3.arcgis.com/FsUrhUGHe9VfghT8/ArcGIS/rest/services/Weekly_HCPH_VDU_Site_Update_0/FeatureServer/0
+    # end_date = site["attributes"].get("close_date")
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/tx/harriscounty_gov/normalize.py:116]: Can you remove the unused commented out code

**Review by bryanculbertson — CHANGES_REQUESTED** [review]

**Comment by bryanculbertson** [comment]
Can you rebase to the latest and remove your changes to poetry lock in this PR?

**Comment by MoralCode** [comment]
@bryanculbertson yeha i can do my best to remove the unnecessary changes to the lockfiles, but this branch requires an update that i made in the latest version of my `parse-opening-hours` library that it needs to function and poetry seems to keep pulling in additional updates when installing it :/ any suggestions for how to proceed?

**Comment by bryanculbertson** [comment]
@MoralCode Ah, I see. Lets upgrade `parse-opening-hours` separately. I would like all packages updates to go via dependabot if we can as a way of tracking them

**Comment by bryanculbertson** [comment]
@MoralCode Processing the library update in this PR:  https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/819

**Comment by MoralCode** [comment]
ah, did not realize dependabot did normal updates and not  just security updates. neat

**Comment by bryanculbertson** [comment]
@MoralCode Library update has processed. You can now rebase this PR.

**Comment by MoralCode** [comment]
@bryanculbertson finally done + rebased. had to make some changes to the vaccine type detection since the initial version was only set for pfizer so that may need a review since it was added just now.

other than that all is rebased and the commits that change the lockfile are totally gone

**Review by bryanculbertson — APPROVED** [review]
🎉

**Pull request merged**